File: AdventOfCode/2025/10/2.py
import fileinput
from z3 import *

res = 0
for line in fileinput.input():
    words = line.split(' ')
    buttons = [list(map(int, ws[1:-1].split(','))) for ws in words[1:-1]]
    voltages = list(map(int, words[-1][1:-2].split(',')))

    opt = z3.Optimize()

    # Variables
    vs = [Int(f'v_{i}') for i in range(len(buttons))]
    for v in vs:
        opt.add(v >= 0)
    opt.minimize(sum(vs))

    # Constraints
    for i in range(len(voltages)):
        ts = []
        for j in range(len(buttons)):
            if i in buttons[j]:
                ts.append(vs[j])
        opt.add(sum(ts) == voltages[i])

    # Solve
    assert opt.check()
    for v in vs:
        res += opt.model()[v].as_long()
print(res)

File: flipflop/2025/7/2.cc
#include "../utils.cc"

ll getAmt(int r, int c, int w) {
    if(!r && !c && !w) return 1;
    return (r ? getAmt(r - 1, c, w) : 0) + 
           (c ? getAmt(r, c - 1, w) : 0) + 
           (w ? getAmt(r, c, w - 1) : 0);
}

void solve() {
    ll res = 0;
    getInp([&](auto& in, int) {
        int r, c; in >> r >> c;
        res += getAmt(r - 1, c - 1, r - 1);
    });
    cout << res << endl;
}

